tools/RepositoryManagement/DevHome.RepositoryManagement/ViewModels/RepositoryManagementMainPageViewModel.cs

```diff
@@ -68,13 +150,21 @@ public void NavigateToCloneRepositoryExpirence()
     }
 
     [RelayCommand]
-    public void LoadRepositories()
+    public async Task LoadRepositories()
     {
-        _items.Clear();
-        Items.Clear();
-        var repositoriesFromTheDatabase = _dataAccessService.GetRepositories();
-        ConvertToLineItems(repositoriesFromTheDatabase).ForEach(x => _items.Add(x));
-        _items.Where(x => x.IsHiddenFromPage == false).ToList().ForEach(x => Items.Add(x));
+        // TODO: Spinning progress ring when loading repositories.
+        AreFilterAndSortEnabled = false;
+        _allRepositoriesFromTheDatabase = _dataAccessService.GetRepositories();
+        _allRepositoriesFromTheDatabase = await AssignSourceControlId(_allRepositoriesFromTheDatabase);
+
+        var repositoriesWithCommits = AssignCommitToRepository(_allRepositoriesFromTheDatabase);
+
+        _allLineItems.Clear();
+        LineItemsToDisplay.Clear();
+        ConvertToLineItems(repositoriesWithCommits).ForEach(x => _allLineItems.Add(x));
```

Comment:
can't use C# assignment?

        _allLineItems = ConvertToLineItems(repositoriesWithCommits);

Reply:
and,

    LineItemsToDisplay = HideFilterAndSort(_allLineItems).Where(x => x.IsHiddenFromPage == false).ToList()

Reply:
Very correct.  I didn't think to re-assign the collections.